recipes/exiv2/all/conanfile.py

```diff
@@ -0,0 +1,119 @@
+import os
+from conans import ConanFile, CMake, tools
+
+required_conan_version = ">=1.33.0"
+
+
+class Exiv2Conan(ConanFile):
+    name = "exiv2"
+    description = "Exiv2 is a C++ library and a command-line utility " \
+                  "to read, write, delete and modify Exif, IPTC, XMP and ICC image metadata."
+    license = "GPL-2.0"
+    topics = ("conan", "image", "exif", "xmp")
+    homepage = "https://www.exiv2.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_png": [True, False],
+        "with_xmp": [True, False],
+        "with_curl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_png": True,
+        "with_xmp": True,
+        "with_curl": False,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libiconv/1.16")
+        if self.options.with_png:
+            self.requires("libpng/1.6.37")
+        if self.options.with_xmp:
+            self.requires("expat/2.4.1")
+        if self.options.with_curl:
+            self.requires("libcurl/7.64.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["EXIV2_BUILD_SAMPLES"] = False
+        self._cmake.definitions["EXIV2_BUILD_EXIV2_COMMAND"] = False
+        self._cmake.definitions["EXIV2_ENABLE_PNG"] = self.options.with_png
+        self._cmake.definitions["EXIV2_ENABLE_XMP"] = self.options.with_xmp
+        self._cmake.definitions["EXIV2_ENABLE_NLS"] = False
```

Comment:
We should add a TODO here.
Or maybe a `with_nls` option that defaults to `False` and raises if `True`?
I don't know what the best practice is in this cases.

Reply:
nls stands for "native language support". So for some type of internationalization.
It's often used by executables, not the libraries.
I'm fine with force disabling nls.

Reply:
I don't know, can we elaborate if it's really used for something in library? e.g. error messages or whatever?

Reply:
Only `gdbm` recipes has an option for nls. Checking my test_package logs, most of time, they enable nls, but test package is an application.